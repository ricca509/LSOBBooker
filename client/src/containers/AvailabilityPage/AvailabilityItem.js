import React from 'react';
import moment from 'moment';

import AvailabilityTimes from './AvailabilityTimes';

export default ({ availability }) => (
  <dt>
    {moment(availability.date).format('MMMM Do YYYY')}
    <AvailabilityTimes times={availability.availability} />
  </dt>
);
