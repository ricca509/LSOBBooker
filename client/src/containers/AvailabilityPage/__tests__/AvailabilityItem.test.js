import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AvailabilityItem from '../AvailabilityItem';

describe('AvailabilityItem', () => {
  it('renders a formatted date and a list of AvailabilityTimes components', () => {
    const availability = {
      date: "2017-01-04T13:09:55.189",
      times: [ '12:00:00', '20:00:00' ]
    };

    const wrapper = shallow(<AvailabilityItem availability={availability} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('does not render when there is no availability', () => {
    const availability = {
      date: "2017-01-04T13:09:55.189",
      times: []
    };

    const wrapper = shallow(<AvailabilityItem availability={availability} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
