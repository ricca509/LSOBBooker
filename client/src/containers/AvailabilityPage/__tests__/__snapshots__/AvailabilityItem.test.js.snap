exports[`AvailabilityItem does not render when there is no availability 1`] = `null`;

exports[`AvailabilityItem renders a formatted date and a list of AvailabilityTimes components 1`] = `
<tr>
  <td>
    Wednesday, January 4th 2017
  </td>
  <td>
    <Component
      times={
        Array [
          "12:00:00",
          "20:00:00",
        ]
      } />
  </td>
</tr>
`;
